docs/gfx.md

```diff
@@ -0,0 +1,124 @@
+ZJS API for GFX
+===============
+
+* [Introduction](#introduction)
+* [Web IDL](#web-idl)
+* [API Documentation](#api-documentation)
+* [Sample Apps](#sample-apps)
+
+Introduction
+------------
+The GFX module provides a generic way to create pixel buffers, which can then
+be displayed on a display of some kind.  A Javascript method for initializing
+the screen, and drawing a data buffer are required to use it.
+See module/ST7735.js and samples/SPI_Screen.js for an example.
+
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.
+
+```javascript
+// require returns a GFX object
+// var gfx = require('gfx');
+
+[NoInterfaceObject]
+interface GFX {
+    GFXObj init(screen width, screen height, init screen function,
+                draw function, optional this);
+};
+
+[NoInterfaceObject]
+interface GFXObj {
+    fillRect(number x coord, number y coord, number width, number height,
+             char array color);
+    drawPixel(number x coord, number y coord, char array color);
+    drawLine(number x0 coord, number y0 coord, number x1 coord,
```

Comment:
Doing something like this, would probably make much more sense and it brings more familiarity:

```
interface CanvasDrawPath {
  // path API (see also CanvasPath)
  void beginPath();
  void fill(optional CanvasFillRule fillRule = "nonzero");
  void fill(Path2D path, optional CanvasFillRule fillRule = "nonzero");
...

interface CanvasPath {
  // shared path API methods
  void closePath();
  void moveTo(unrestricted double x, unrestricted double y);
  void lineTo(unrestricted double x, unrestricted double y);
...
Path2D implements CanvasPath;
```

Reply:
Perhaps in the future, but that's more advanced than I was trying to accomplish for right now.